### ISSUE TITLE
[BUG] The .d.ts declaration file is not valid

### ISSUE THREAD

**Issue by zcbenz** [lenML/tokenizers#4]
Setup a simple TypeScript project adding this module as dependency, import this module, and running `tsc` results in following error:

```
node_modules/@lenml/tokenizers/dist/main.d.ts:308:69 - error TS2339: Property 'progress_callback' does not exist on type 'undefined'.

308     static from_pretrained(pretrained_model_name_or_path: string, { progress_callback, config, cache_dir, local_files_only, revision, legacy, }?: PretrainedTokenizerOptions): Promise<PreTrainedTokenizer>;
                                                                        ~~~~~~~~~~~~~~~~~
```

Which seems to be caused by `PretrainedOptions` from `hub.js` being `undefined`.

Copying the definition from upstream should be able to fix it, but I'm not sure if there are other issues might occur:  https://github.com/xenova/transformers.js/blob/f43d3dd348fd7b293008802590bb3a1afa218dc7/src/utils/hub.js#L14-L27`

**Comment by zhzLuke96** [comment]
Thanks for the feedback!

In addition to the issue you mentioned, I found two other type issues and attempted to fix them as well. You can try the new version @1.1.2.

By the way, these type issues might originate from `transformers.js`. There are actually some simple workarounds, like adding `xxx as any;` in the appropriate places.

Fixed in 3e08f12dd02ad86ac4136b5909305a6981d15de9.

**Comment by zcbenz** [comment]
The new version works perfectly, thanks a lot!
